### DISCUSSION TITLE
HMS-400BM identified as "Unknown" in the inverters.

### DISCUSSION THREAD

**Discussion by creaimmo** [tbnobody/OpenDTU#2458] [Q&A]
Hello,

I am trying to set up a microinverter with the reference HMS-400BM. It is correctly recognized with a Hoymiles serial number, but once added, it is marked as type "Unknown."

**Comment by tbnobody** [comment]
Then please Tell us the serial Number (at least the first six digits) otherwise it's not possible to further investigate this

**Reply by stefan123t** [reply]
Please also tell us your country of residence and the purchase. I believe we already had one HMS-400BM previously and the idea was that this model is available in some countries only ?

@creaimmo where did you buy this inverter from (dealer / manufacturer) and what is your country of residence?
We still have too little info about this model.

There has been another HMS-400BM in #2170 owned by @tt-why from the company Beem in France.

You may also want to double check closed issues #1540 and #1190 for details on the ChannelChangeCommand.

**Comment by creaimmo** [comment]
the SN: 1400A01
Buy from BEEM in France

**Comment by creaimmo** [comment]
![Onduleur jpg](https://github.com/user-attachments/assets/2063eabf-3e8a-452e-9b67-b426d920f69c)
![OpenDTU-param-onduleur](https://github.com/user-attachments/assets/5546fa6f-2d55-460d-a393-efa1fe840b7f)

**Reply by stefan123t** [reply]
Please add the Hardware and Firmware Version numbers and other details from the Inverter Info (MCU icon) page.

I think tbnobody will add the Beem OEM inverters to the code once we have all the details and you have been able to confirm that most of the data retrieved from the inverter looks plausible?

eg like this
![image](https://github.com/user-attachments/assets/3ecfe8d3-c139-45dd-b924-a01bc179afe8)

**Comment by geronimo-78** [comment]
Where to find this information knowing that on the direct page I do not have an MCU icon
![OpenDTU-direct](https://github.com/user-attachments/assets/82314e29-a427-4f94-b6fc-ea9c9adbd66a)

**Reply by ghost** [reply]
Unfortunately it's not that nice. Otherwise it's here:

![Screenshot 2024-12-16 181955](https://github.com/user-attachments/assets/7590cf3c-f580-41bb-a277-525c4228327c)

**Reply by tbnobody** [reply]
The problem is, that the mentioned serial number 1400xxxxx is not yet assigned to a inverter type. Without this, it gets not read. 
The 1400xxxxx is already known and I will add it in the next release. (Then the MCU button should appear as well)

**Comment by geronimo-78** [comment]
This explains the problem. When do you plan to make this new version available?
Sincerely

**Comment by creaimmo** [comment]
Top! We are waiting for the next version

**Reply by stefan123t** [reply]
There is the idea to select a matching model from the known list of models when the actual model prefix is still unknown.

You can try it with either `1124` (v1) or `1125` (v2) as a workaround using the currently implemented models:
https://github.com/tbnobody/OpenDTU/issues/2412#issuecomment-2492553162

This discussion may be closed as duplicate of #2412

**Comment by geronimo-78** [comment]
Below the information of the DTU after modification of the SN

![OpenDTU-info-onduleur copie](https://github.com/user-attachments/assets/5571290c-f244-4cb4-bbb6-777bc6dbb47e)
